### PR TITLE
fixed issue #2 - paragraphs offset

### PR DESCRIPTION
**Pull request by adrianmarinwork**
fixed issue #2 - paragraphs offset

Okay, I'm going to add the bible to explain what I have done and why.

- What have I done?

I have added the attribute `onclone `to the `html2canvas` instance, this callback triggers when the library has finished of cloning the content to the canvas. Then we can edit the content of the document without changing/affecting the actual content of the page, thanks to that I added `position: relative` to the containers that contain the text from the messages and moved it up with `top: -8px`. 
I've removed the `copy code` button that appears at the top right corner of the code blocks (you can see an example in one of the screenshots I sent you in our issue conversation) because the text was moving down too for some reason, as I can't move that text up, as the button is useless in a PNG or PDF, I have removed it from the result.

- Why have I done this to fix the issue?

This problem of having the text with a small offset above it is something common in the html2canvas library ([same issue](https://github.com/niklasvh/html2canvas/issues/2775)), as this is not our page and I can't change the content of it before rendering the PNG, I decided to do it with the callback `onclone`.

Well, If you don't like something about the fix or want to ask/suggest me something, go ahead!

### CODE REVIEW

File: src/content_script.js [hunk 1]
```diff
@@ -70,9 +70,29 @@ function downloadThread({ as = Format.PNG } = {}) {
...
+        //Delete copy button from code blocks
+        let listOfCopyBtns = cloneDoc.querySelectorAll('button.flex');
+
+        for (let i = 0, leng = listOfCopyBtns.length; i < leng; i++) {
+          let button = listOfCopyBtns[i];
+          button.remove();
+        }
+      }
+    }
+  ).then(async function (canvas) {
```
Comment (liady) [inline src/content_script.js:95]: Looks good! What do you think about using `forEach`?

```suggestion
  html2canvas(elements.thread, {
    letterRendering: true,
    onclone: function (cloneDoc) {
      // Make small fix of position to all the text containers
      let listOfTexts = cloneDoc.getElementsByClassName("min-h-[20px]");
      Array.from(listOfTexts).forEach((text) => {
        text.style["position"] = "relative";
        text.style["top"] = "-8px";
      });
      // Delete copy button from code blocks
      let listOfBtns = cloneDoc.querySelectorAll("button.flex");
      Array.from(listOfBtns).forEach((btn) => btn.remove());
    },
  }).then(async function (canvas) {
```

**Review by liady — CHANGES_REQUESTED** [review]

File: src/content_script.js (hunk 1 above)
Comment (adrianmarinwork) [reply src/content_script.js]: Done!

**Comment by adrianmarinwork** [comment]
Done, I've converted the loops to foreach as you requested.

**Comment by liady** [comment]
Thanks!

**Pull request merged**